server/src-lib/Hasura/GraphQL/Transport/HTTP.hs

```diff
@@ -26,15 +28,56 @@ runGQ
   -> HTTP.Manager
   -> [N.Header]
   -> GraphQLRequest
-  -> BL.ByteString -- this can be removed when we have a pretty-printer
   -> m EncJSON
-runGQ pool isoL userInfo sqlGenCtx sc manager reqHdrs req rawReq = do
+runGQ pool isoL userInfo sqlGenCtx sc manager reqHdrs req = do
   execPlan <- E.getExecPlan userInfo sc req
   case execPlan of
     E.GExPHasura gCtx rootSelSet ->
       runHasuraGQ pool isoL userInfo sqlGenCtx gCtx rootSelSet
-    E.GExPRemote rsi opDef  ->
-      E.execRemoteGQ manager userInfo reqHdrs rawReq rsi opDef
+    E.GExPRemote rsi _ rs ->
+      E.execRemoteGQ manager userInfo reqHdrs req rsi rs
+    E.GExPMixed plans ->
+      runMixedGQ pool isoL userInfo sqlGenCtx manager reqHdrs plans
+
+
+runMixedGQ
+  :: (MonadIO m, MonadError QErr m)
+  => Q.PGPool
+  -> Q.TxIsolation
+  -> UserInfo
+  -> SQLGenCtx
+  -> HTTP.Manager
+  -> [N.Header]
+  -> [E.GQExecPlan]
+  -> m EncJSON
+runMixedGQ pool isoL userInfo sqlGenCtx manager reqHdrs plans = do
+  resSet <- forM plans $ \case
+    E.GExPHasura gCtx rootSelSet ->
+      runHasuraGQ pool isoL userInfo sqlGenCtx gCtx rootSelSet
+    E.GExPRemote rsi newq rs ->
+      E.execRemoteGQ manager userInfo reqHdrs newq rsi rs
+    E.GExPMixed _ ->
+      throw500 "internal-unexpected: mixed plan is nested in mixed plan"
+
+  let interimResBS = map encJToLBS resSet
+  interimRes <- forM interimResBS $ \res -> do
+    let x = J.decode res :: (Maybe J.Object)
+    onNothing x $ throw500 "could not parse response as JSON"
+
+  let datas = onlyObjs $ mapMaybe (Map.lookup "data") interimRes
+      errs  = mapMaybe (Map.lookup "errors") interimRes
```

Comment:
errs should be Maybe type so that you don't get the `errors` key if the errors array is empty